lib/internal/modules/esm/hooks.js

```diff
@@ -590,16 +590,17 @@ class HooksProxy {
     if (response.message.status === 'never-settle') {
       return new Promise(() => {});
     }
-    if (response.message.status === 'error') {
-      if (response.message.body == null) throw response.message.body;
-      if (response.message.body.serializationFailed || response.message.body.serialized == null) {
+    const { status, body } = response.message;
+    if (status === 'error') {
+      if (body == null || typeof body !== 'object') throw body;
```

Comment:
this would rethrow a function too.
```suggestion
      if (body == null || Object(body) !== body) throw body;
```

(alternatively, this needs to check both functions and symbols as well)

Reply:
I don't see why, throwing a function or a symbol would be perfectly valid – and the structured clone algorithm do not support them anyway, so there not a chance that `body` would be a function nor a symbol.

Reply:
ahh ok this `body` can only come from a structured clone? in that case, why are functions and symbols checked in lib/internal/modules/esm/worker.js?

Reply:
`hooks.js` is run on the main thread, it's the output of the structured clone. `worker.js` runs on the loader thread, it's the input of the structured clone (meaning we need to serialize functions and symbols there so it can be passed to the main thread)

Reply:
Makes sense. It still might be nice to have the checks be the same, or at least have a comment explaining the difference?

Reply:
I'm tempted to leave it as is, so I don't have to fight with CI. Feel free to send a follow up PR though, I'll make sure to approve it and fast track it.

Reply:
Sure, which approach do you prefer?

Reply:
I think a comment (or two?) should be enough – and likely the fastest at runtime